### DISCUSSION TITLE
Leaf Map not accepting an argument

### DISCUSSION THREAD

**Discussion by Diego-asutralgis** [opengeos/segment-geospatial#364] [Q&A]
Hello, I had a question about a leafmap error that happens to me when I try to reproduce the notebook, (Map.add() got an unexpected keyword argument 'index'), I created a new VE environment to run the code and had to  update some libraries do to conflicts, If anyone could shed some light on this issue I´d be glad, thanks in advance!

**Comment by giswqs** [comment]
Please a complete code snippet that can reproduce the issue.

**Reply by Diego-asutralgis** [reply]
![image](https://github.com/user-attachments/assets/fdd236e9-e865-4788-9719-615ccc6f0b59)

**Reply by Diego-asutralgis** [reply]
this is what i can show besides the library that I´m currently using

# Name                    Version                   Build  Channel
addict                    2.4.0                    pypi_0    pypi
affine                    2.4.0                    pypi_0    pypi
aniso8601                 9.0.1                    pypi_0    pypi
annotated-types           0.7.0                    pypi_0    pypi
antlr4-python3-runtime    4.9.3                    pypi_0    pypi
anyio                     4.6.2.post1              pypi_0    pypi
anywidget                 0.9.13                   pypi_0    pypi
asttokens                 2.4.1              pyhd8ed1ab_0    conda-forge
attrs                     24.2.0                   pypi_0    pypi
beautifulsoup4            4.12.3                   pypi_0    pypi
blas                      1.0                         mkl
blinker                   1.8.2                    pypi_0    pypi
blosc                     1.21.3               h6c2663c_0
boost-cpp                 1.82.0               h59b6b97_2
bqplot                    0.12.43                  pypi_0    pypi
branca                    0.8.0                    pypi_0    pypi
bzip2                     1.0.8                h2bbff1b_6
ca-certificates           2024.9.24            haa95532_0
cachelib                  0.9.0                    pypi_0    pypi
cachetools                5.5.0                    pypi_0    pypi
cairo                     1.16.0               hc68a040_5
certifi                   2024.8.30                pypi_0    pypi
cfitsio                   3.470                h2bbff1b_7
charset-normalizer        3.4.0                    pypi_0    pypi
click                     8.1.7                    pypi_0    pypi
click-plugins             1.1.1                    pypi_0    pypi
cligj                     0.7.2                    pypi_0    pypi
color-operations          0.1.6                    pypi_0    pypi
colorama                  0.4.6              pyhd8ed1ab_0    conda-forge
colour                    0.1.5                    pypi_0    pypi
comm                      0.2.2              pyhd8ed1ab_0    conda-forge
contourpy                 1.3.0                    pypi_0    pypi
cycler                    0.12.1                   pypi_0    pypi
debugpy                   1.8.7           py311hda3d55a_0    conda-forge
decorator                 5.1.1              pyhd8ed1ab_0    conda-forge
duckdb                    1.1.3                    pypi_0    pypi
exceptiongroup            1.2.2              pyhd8ed1ab_0    conda-forge
executing                 2.1.0              pyhd8ed1ab_0    conda-forge
expat                     2.6.3                h5da7b33_0
filelock                  3.16.1                   pypi_0    pypi
fiona                     1.10.1                   pypi_0    pypi
flask                     3.0.3                    pypi_0    pypi
flask-caching             2.3.0                    pypi_0    pypi
flask-cors                5.0.0                    pypi_0    pypi
flask-restx               1.3.0                    pypi_0    pypi
folium                    0.18.0                   pypi_0    pypi
fontconfig                2.14.1               hb33846d_3
fonttools                 4.54.1                   pypi_0    pypi
freetype                  2.12.1               ha860e81_0
freexl                    2.0.0                hd7a5696_0
fsspec                    2024.10.0                pypi_0    pypi
gdal                      3.6.2           py311h4e7b5b2_6
gdown                     5.2.0                    pypi_0    pypi
geojson                   3.1.0                    pypi_0    pypi
geopandas                 1.0.1                    pypi_0    pypi
geos                      3.8.0                h33f27b4_0
geotiff                   1.7.0                h4545760_3
glib                      2.78.4               hd77b12b_0
glib-tools                2.78.4               hd77b12b_0
h11                       0.14.0                   pypi_0    pypi
hdf4                      4.2.13               h712560f_2
hdf5                      1.12.1               h51c971a_3
httpcore                  1.0.6                    pypi_0    pypi
httpx                     0.27.2                   pypi_0    pypi
huggingface-hub           0.26.2                   pypi_0    pypi
hydra-core                1.3.2                    pypi_0    pypi
icc_rt                    2022.1.0             h6049295_2
idna                      3.10                     pypi_0    pypi
imageio                   2.36.0                   pypi_0    pypi
importlib-metadata        8.5.0              pyha770c72_0    conda-forge
importlib-resources       6.4.5                    pypi_0    pypi
intel-openmp              2023.1.0         h59b6b97_46320
iopath                    0.1.10                   pypi_0    pypi
ipyevents                 2.0.2                    pypi_0    pypi
ipyfilechooser            0.6.0                    pypi_0    pypi
ipykernel                 6.29.5             pyh4bbf305_0    conda-forge
ipyleaflet                0.19.2                   pypi_0    pypi
ipympl                    0.9.4                    pypi_0    pypi
ipython                   8.29.0             pyh7428d3b_0    conda-forge
ipython-genutils          0.2.0                    pypi_0    pypi
ipytree                   0.2.2                    pypi_0    pypi
ipyvue                    1.11.1                   pypi_0    pypi
ipyvuetify                1.10.0                   pypi_0    pypi
ipywidgets                8.1.5                    pypi_0    pypi
itsdangerous              2.2.0                    pypi_0    pypi
jedi                      0.19.1             pyhd8ed1ab_0    conda-forge
jinja2                    3.1.4           py311haa95532_1
joblib                    1.4.2                    pypi_0    pypi
jpeg                      9e                   h827c3e9_3
jsonschema                4.23.0                   pypi_0    pypi
jsonschema-specifications 2024.10.1                pypi_0    pypi
jupyter-leaflet           0.19.2                   pypi_0    pypi
jupyter_client            8.6.3              pyhd8ed1ab_0    conda-forge
jupyter_core              5.7.2           py311h1ea47a8_0    conda-forge
jupyterlab-widgets        3.0.13                   pypi_0    pypi
jupyterlab_widgets        3.0.10          py311haa95532_0
kealib                    1.5.0                hde4a422_1
kiwisolver                1.4.7                    pypi_0    pypi
krb5                      1.21.3               hdf4eb48_0    conda-forge
lazy-loader               0.4                      pypi_0    pypi
lcms2                     2.12                 h83e58a3_0
leafmap                   0.38.16                  pypi_0    pypi
lerc                      3.0                  hd77b12b_0
libboost                  1.82.0               h3399ecb_2
libcurl                   8.9.1                h0416ee5_0
libdeflate                1.17                 h2bbff1b_1
libffi                    3.4.4                hd77b12b_1
libgdal                   3.6.2                h0e70117_6
libglib                   2.78.4               ha17d25a_0
libiconv                  1.16                 h2bbff1b_3
libkml                    1.3.0                h63940dd_7
libnetcdf                 4.8.1                h6685c40_4
libpng                    1.6.39               h8cc25b3_0
libpq                     12.15                h906ac69_0
libsodium                 1.0.20               hc70643c_0    conda-forge
libspatialite             5.1.0                h25d3e1c_1
libssh2                   1.11.0               h291bd65_0
libtiff                   4.5.1                hd77b12b_0
libwebp-base              1.3.2                h3d04722_1
libxml2                   2.13.1               h24da03e_2
libzip                    1.8.0                h289538f_1
localtileserver           0.10.5                   pypi_0    pypi
lz4-c                     1.9.4                h2bbff1b_1
markupsafe                3.0.2                    pypi_0    pypi
matplotlib                3.9.2                    pypi_0    pypi
matplotlib-inline         0.1.7              pyhd8ed1ab_0    conda-forge
minizip                   4.0.3                hb68bac4_0
mkl                       2023.1.0         h6b88ed4_46358
mkl-service               2.4.0           py311h2bbff1b_1
mkl_fft                   1.3.11          py311h827c3e9_0
mkl_random                1.2.8           py311hea22821_0
morecantile               5.4.2                    pypi_0    pypi
mpmath                    1.3.0                    pypi_0    pypi
nest-asyncio              1.6.0              pyhd8ed1ab_0    conda-forge
networkx                  3.4.2                    pypi_0    pypi
numexpr                   2.10.1                   pypi_0    pypi
numpy                     2.1.3                    pypi_0    pypi
omegaconf                 2.3.0                    pypi_0    pypi
opencv-python-headless    4.10.0.84                pypi_0    pypi
openjpeg                  2.5.2                hae555c5_0
openssl                   3.3.2                h2466b09_0    conda-forge
packaging                 24.1               pyhd8ed1ab_0    conda-forge
pandas                    2.2.3                    pypi_0    pypi
parso                     0.8.4              pyhd8ed1ab_0    conda-forge
patool                    3.0.3                    pypi_0    pypi
pcre2                     10.42                h0ff8eda_1
pickleshare               0.7.5                   py_1003    conda-forge
pillow                    11.0.0                   pypi_0    pypi
pip                       24.2            py311haa95532_0
pixman                    0.40.0               h2bbff1b_1
platformdirs              4.3.6              pyhd8ed1ab_0    conda-forge
plotly                    5.24.1                   pypi_0    pypi
poppler                   22.12.0              h0bf3bde_3
poppler-data              0.4.11               haa95532_1
portalocker               2.10.1                   pypi_0    pypi
proj                      9.3.1                ha107b6e_0
prompt-toolkit            3.0.48             pyha770c72_0    conda-forge
psutil                    6.1.0           py311he736701_0    conda-forge
psygnal                   0.11.1                   pypi_0    pypi
pure_eval                 0.2.3              pyhd8ed1ab_0    conda-forge
pycocotools               2.0.8                    pypi_0    pypi
pydantic                  2.9.2                    pypi_0    pypi
pydantic-core             2.23.4                   pypi_0    pypi
pygments                  2.18.0             pyhd8ed1ab_0    conda-forge
pyogrio                   0.10.0                   pypi_0    pypi
pyparsing                 3.2.0                    pypi_0    pypi
pyproj                    3.7.0                    pypi_0    pypi
pyshp                     2.3.1                    pypi_0    pypi
pysocks                   1.7.1                    pypi_0    pypi
pystac                    1.11.0                   pypi_0    pypi
pystac-client             0.8.5                    pypi_0    pypi
python                    3.11.9               he1021f5_0
python-box                7.2.0                    pypi_0    pypi
python-dateutil           2.9.0.post0              pypi_0    pypi
python_abi                3.11                    2_cp311    conda-forge
pytz                      2024.2                   pypi_0    pypi
pywin32                   308                      pypi_0    pypi
pyyaml                    6.0.2                    pypi_0    pypi
pyzmq                     26.2.0          py311h484c95c_3    conda-forge
qhull                     2020.2               h59b6b97_2
rasterio                  1.4.2                    pypi_0    pypi
referencing               0.35.1                   pypi_0    pypi
regex                     2024.11.6                pypi_0    pypi
requests                  2.32.3                   pypi_0    pypi
rio-cogeo                 5.3.6                    pypi_0    pypi
rio-tiler                 7.2.0                    pypi_0    pypi
rioxarray                 0.18.0                   pypi_0    pypi
rpds-py                   0.21.0                   pypi_0    pypi
safetensors               0.4.5                    pypi_0    pypi
sam2                      0.4.1                    pypi_0    pypi
scikit-image              0.24.0                   pypi_0    pypi
scikit-learn              1.5.2                    pypi_0    pypi
scipy                     1.14.1                   pypi_0    pypi
scooby                    0.10.0                   pypi_0    pypi
segment-anything-hq       0.3                      pypi_0    pypi
segment-anything-py       1.0.1                    pypi_0    pypi
segment-geospatial        0.12.2                   pypi_0    pypi
server-thread             0.2.0                    pypi_0    pypi
setuptools                75.1.0          py311haa95532_0
shapely                   2.0.6                    pypi_0    pypi
six                       1.16.0             pyh6c4a22f_0    conda-forge
sniffio                   1.3.1                    pypi_0    pypi
soupsieve                 2.6                      pypi_0    pypi
sqlite                    3.45.3               h2bbff1b_0
stack-data                0.6.3                    pypi_0    pypi
stack_data                0.6.2              pyhd8ed1ab_0    conda-forge
sympy                     1.13.1                   pypi_0    pypi
tbb                       2021.8.0             h59b6b97_0
tenacity                  9.0.0                    pypi_0    pypi
threadpoolctl             3.5.0                    pypi_0    pypi
tifffile                  2024.9.20                pypi_0    pypi
tiledb                    2.3.3                hd8964de_3
timm                      1.0.11                   pypi_0    pypi
tk                        8.6.14               h0416ee5_0
tomli                     2.0.2                    pypi_0    pypi
torch                     2.5.1                    pypi_0    pypi
torchvision               0.20.1                   pypi_0    pypi
tornado                   6.4.1           py311he736701_1    conda-forge
tqdm                      4.67.0                   pypi_0    pypi
traitlets                 5.14.3             pyhd8ed1ab_0    conda-forge
traittypes                0.2.1           py311haa95532_0
typing_extensions         4.12.2             pyha770c72_0    conda-forge
tzdata                    2024.2                   pypi_0    pypi
ucrt                      10.0.22621.0         h57928b3_1    conda-forge
uriparser                 0.9.7                h2bbff1b_0
urllib3                   2.2.3                    pypi_0    pypi
uvicorn                   0.32.0                   pypi_0    pypi
vc                        14.40                h2eaa2aa_1
vc14_runtime              14.40.33810         hcc2c482_22    conda-forge
vs2015_runtime            14.40.33810         h3bf8584_22    conda-forge
wcwidth                   0.2.13             pyhd8ed1ab_0    conda-forge
werkzeug                  3.1.2                    pypi_0    pypi
wheel                     0.44.0          py311haa95532_0
whitebox                  2.3.5                    pypi_0    pypi
whiteboxgui               2.3.0                    pypi_0    pypi
widgetsnbextension        4.0.13                   pypi_0    pypi
xarray                    2024.10.0                pypi_0    pypi
xerces-c                  3.2.4                hd77b12b_1
xyzservices               2024.9.0                 pypi_0    pypi
xz                        5.4.6                h8cc25b3_1
zeromq                    4.3.5                ha9f60a1_6    conda-forge
zipp                      3.20.2             pyhd8ed1ab_0    conda-forge
zlib                      1.2.13               h8cc25b3_1
zstd                      1.5.6                h8880b57_0
